src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,53 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+
+    public static final int FIGURE_COUNT = 5;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int figureNumber = random.nextInt(FIGURE_COUNT);
+        // generate a specific figure based on the `figureNumber` value
+        switch (figureNumber) {
+            case 0:
+                Circle circle = new Circle();
+                circle.setColor(colorSupplier.getRandomColor());
```

Comment:
You can create local variable color = colorSupplier.getRandomColor(); and use it in each case

Reply:
Yes I can but it will be the same color for each case